### PR TITLE
Add functionality for favorite movies -> Add favorite filed in popular movies

### LINKED ISSUE #132: Add functionality for favorite movies
## Description
I want to add a feature of adding a movie to my favorite.

## What I should add:
- Placeholder if the list will be empty
- I will add new value to the movie data class flag `isFavorite`
- I will use update method on room database
- I should consider use flow if updating has any problem
- Icon heart (favorite) to each element in the list of movies
- Method onClick updating movie to favorite or removing from favorite
- Consider the case when the User adds a movie to his favorite and he hasn't fetched details for the movie what am I supposed to do with it? I should always try to fetch data or only when the user clicks on the movie on the list of movies or favorite movies.   

- [x] Add heart icon to movie list
- [x] Add favorite filed in popular movies
- [x] Scheme to change color if movie is favorite 
- [x] Update isFavorite filed in database
- [x] Get favorite movies from database
- [ ] Create UI for favourite movies

### PR DESCRIPTION
**Pull request by Nataniel-Antosik**
## DONE

### CODE REVIEW

File: app/src/main/kotlin/com/antosik/benchproject/app/popular/movies/entity/Movie.kt [hunk 1]
```diff
@@ -8,6 +8,7 @@ data class Movie(
     val rating: Double,
     val releaseDate: String,
     val imageUrl: String,
+    val favorite: Boolean,
```
Comment (BorysChajdas) [inline app/src/main/kotlin/com/antosik/benchproject/app/popular/movies/entity/Movie.kt]: `isFavorite`

File: data/src/main/kotlin/com/antosik/benchproject/data/popular/movies/entity/database/MovieEntity.kt [hunk 2]
```diff
@@ -12,6 +12,7 @@ internal data class MovieEntity(
     val rating: Double,
     val releaseDate: String,
     val imagePath: String,
+    val favorite: Boolean,
```
Comment (BorysChajdas) [inline data/src/main/kotlin/com/antosik/benchproject/data/popular/movies/entity/database/MovieEntity.kt]: `isFavorite`

File: domain/src/main/kotlin/com/antosik/benchproject/domain/popular/movies/entity/MovieModel.kt [hunk 3]
```diff
@@ -6,4 +6,5 @@ data class MovieModel(
     val rating: Double,
     val releaseDate: String,
     val imageUrl: String,
+    val favorite: Boolean,
```
Comment (BorysChajdas) [inline domain/src/main/kotlin/com/antosik/benchproject/domain/popular/movies/entity/MovieModel.kt]: `isFavorite`

**Review by BorysChajdas — APPROVED** [review]

File: domain/src/main/kotlin/com/antosik/benchproject/domain/popular/movies/entity/MovieModel.kt (hunk 3 above)
Comment (Nataniel-Antosik) [reply domain/src/main/kotlin/com/antosik/benchproject/domain/popular/movies/entity/MovieModel.kt]: Done :)

File: data/src/main/kotlin/com/antosik/benchproject/data/popular/movies/entity/database/MovieEntity.kt (hunk 2 above)
Comment (Nataniel-Antosik) [reply data/src/main/kotlin/com/antosik/benchproject/data/popular/movies/entity/database/MovieEntity.kt]: Done :)

File: app/src/main/kotlin/com/antosik/benchproject/app/popular/movies/entity/Movie.kt (hunk 1 above)
Comment (Nataniel-Antosik) [reply app/src/main/kotlin/com/antosik/benchproject/app/popular/movies/entity/Movie.kt]: Done :)

**Pull request merged**